lib/PuppeteerSharp/Launcher.cs

```diff
@@ -127,16 +145,66 @@ public class Launcher
                 var keepAliveInterval = options.KeepAliveInterval;
 
                 _connection = await Connection.Create(browserWSEndpoint, connectionDelay, keepAliveInterval);
+
+                if (options.LogProcess)
+                {
+                    Console.WriteLine($"PROCESS COUNT: {Interlocked.Increment(ref _processCount)}");
+                }
+
                 return await Browser.CreateAsync(_connection, options, KillChrome);
             }
             catch (Exception ex)
             {
-                await ForceKillChrome();
+                ForceKillChrome();
                 throw new Exception("Failed to create connection", ex);
             }
 
         }
 
+        public static async Task TryDeleteDirectory(string folder, int times = 10, TimeSpan? delay = null)
```

Comment:
can we infer the `folder` directory from the `_options.UserDataDir` or `_temporaryUserDataDir`. I don't think we want this method to be used on any directory. If you remove that argument you could also rename it to something like `TryDeleteUserDataDir`

Reply:
I can buy that. But if you provide a UserDataDir, you should call this method explicitly. We would only remove the folder we create automatically.
Agreed?

Reply:
Hmm, didn't think of that. I guess this way is better

Reply:
That's why I exposed it as a public method, If you need it you can use it.

Reply:
If we leave this as generic as it is now. Should we check for `IsChromeClosed`?

Reply:
could it succeed if chrome is open?

Reply:
could it succeed if chrome is open?

Reply:
It would fail after retrying 10 times. So we are calling it after closing chrome we wouldn't fail there. If the user calls it, he would get an exception no matter what we do. If we validate IsChromeClosed he'd get an exception and if not he'd get an exception after the 10th retry.

Reply:
If chrome is open there is no point trying to delete since it will fail

Reply:
How about now @Meir017?